lib/rules/sort-vars.js

```diff
@@ -14,9 +15,90 @@ module.exports = function(context) {
     var configuration = context.options[0] || {},
         ignoreCase = configuration.ignoreCase || false;
 
+
+
     return {
         "VariableDeclaration": function(node) {
-            node.declarations.reduce(function(memo, decl) {
+            var nodeDeclarations;
+            var PatternMapper = {};
+
+             // Adds all leaves to single collection
+            PatternMapper.flattenIntoArray = function flattenIntoArray(obj, collection) {
+                if (Array.isArray(obj)) {
```

Comment:
Actually, I see what you've done here. These should reach just be separate functions. The `patternMapper` , object is unnecessary.